front-end/src/components/CommentFormElement.tsx

```diff
@@ -0,0 +1,85 @@
+import { AxiosWrapper } from '../helpers/AxiosWrapper';
+import { AxiosRequestConfig, AxiosResponse } from 'axios';
+import React, { useState } from "react";
+import { Alert, Button, Card, CardTitle, Form, FormGroup, Input, Modal, ModalBody, ModalFooter, ModalHeader } from "reactstrap";
+import { UserLogin } from "../types/UserLogin";
+import { Author } from '../types/Author';
+import { PostComment } from '../types/PostComment';
+import CommentList from './CommentList';
+
+interface Props {
+  loggedInUser: UserLogin | undefined,
+  postId: string,
+  postAuthor: Author,
+}
+
+/**
+ * Delete Post modal, confirmation to delete post as well as removing from list on success
+ * @param props 
+ */
+export default function CommentFormElement(props: Props) {
+
+  const [showError, setShowError] = useState(false);
+  const [commentContent, setCommentContent] = useState("");
+  const [commenter, setCommenter] = useState<Author | undefined>(undefined);
```

Comment:
This is a note more so for conversation. I personally don't know how to get around not using `undefined`. I used `useState<Author | undefined>` in this project myself, and I think the codebase has become a bit more complicated as a result. I wonder if it's possible to use something like `useState<Author | Promise<Author>>` or similar to prevent working with `undefined` values.

If you have any suggestions about how to go about this let me know :+1:. It's something I'd like to work around when I'm writing this kind of code